src/util.ts

```diff
@@ -166,3 +167,32 @@ export function formatAsUTCISO(
 
   return resultString;
 }
+
+export class PassThroughShim extends PassThrough {
+  private shouldEmitReading = true;
+  private shouldEmitWriting = true;
+
+  constructor() {
+    super();
+  }
+
+  _read(size: number): void {
+    if (this.shouldEmitReading) {
+      this.emit('reading');
+      this.shouldEmitReading = false;
+    }
+    super._read(size);
+  }
+
+  _write(
```

Comment:
You'll need a `_final` handler in addition to this `_write` handler:
- https://nodejs.org/api/stream.html#writable_finalcallback

Reply:
Added.